### PR TITLE
✨(datasources) add SQLite datasource for demonstration instance

### PR DESCRIPTION
**Pull request by quitterie-lcs**
# Purpose

`SQLite` datasource is more convenient to our needs for the demonstration instance of `potsie`. It will replace the `MySQL` database used with `EdX` as there are only a few data stored in.

# Proposal

- [x] Docker-compose for a demonstration version of Grafana
- [x] Makefile command
- [x] Config files for datasource

### CODE REVIEW

File: etc/grafana/provisioning/datasources/sqlite.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: 1
+
+datasources:
+  - name: edx_app_demo
+    type: frser-sqlite-datasource
+    access: proxy
+    isDefault: false
+    editable: false
```
Comment (jmaupetit) [inline etc/grafana/provisioning/datasources/sqlite.yaml:8]: Is it required in development?

File: scripts/users-permissions.sh [hunk 2]
```diff
@@ -4,44 +4,45 @@ set -oe pipefail
...
+PLATFORM_FOLDER_UID=$(${curl} "http://${SERVICE}:3000/api/folders" | jq -r '.[] | select(.title=="platform") | .uid')
 
 # Get "Viewer" permission on "teacher" folder for the "Teacher" team only
 TEACHER_FOLDER_PERMISSION="{\"items\":[{\"teamId\":${TEACHER_TEAM_ID},\"permission\":1}]}"
-${curl} -H "${HEADER}" -d "${TEACHER_FOLDER_PERMISSION}" "http://grafana:3000/api/folders/${TEACHER_FOLDER_UID}/permissions"
+${curl} -H "${HEADER}" -d "${TEACHER_FOLDER_PERMISSION}" "http://${SERVICE}:3000/api/folders/${TEACHER_FOLDER_UID}/permissions"
 
 # Remove permission for "Student" and "Teacher" teams on the "platform" folder
-${curl} -H "${HEADER}" -d "{}" "http://grafana:3000/api/folders/${PLATFORM_FOLDER_UID}/permissions"
+${curl} -H "${HEADER}" -d "{}" "http://${SERVICE}:3000/api/folders/${PLATFORM_FOLDER_UID}/permissions"
```
Comment (jmaupetit) [inline scripts/users-permissions.sh:48]: You don't need to modify this script, right?

File: env.d/demonstration [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Demonstration edx_app datasource
+
+# To uncomment when running the Potsie demonstration 
+EDXAPP_DATASOURCE_DEMO="edx_app_demo"
```
Comment (jmaupetit) [inline env.d/demonstration:4]: I would remove this file. It's a bit too convoluted to me.

File: bin/jsonnet [hunk 4]
```diff
@@ -1,10 +1,16 @@
 #!/usr/bin/env bash
 
+source env.d/demonstration
+
+if [[ $EDXAPP_DATASOURCE_DEMO ]]; then
+  echo -e "\nWARNING !! Demonstration datasource is activated\n"
+fi
+
```
Comment (jmaupetit) [inline bin/jsonnet:8]:
```suggestion
declare -r EDXAPP_DATASOURCE="${EDXAPP_DATASOURCE:-edx_app}"

echo "EDXAPP_DATASOURCE: ${EDXAPP_DATASOURCE}"
```

File: bin/jsonnet [hunk 5]
```diff
@@ -1,10 +1,16 @@
...
 declare DOCKER_USER
 DOCKER_USER="$(id -u):$(id -g)"
 
 # nota bene: we systematically fix local arguments path (containing src/) as
 # the dashboard directory is mounted in the container working directory, e.g.
 # src/dashboards is mounted in /app/dashboards (with /app as the working
 # directory).
-DOCKER_USER=${DOCKER_USER} docker-compose run --rm app jsonnet "${@//src\//}"
+DOCKER_USER=${DOCKER_USER} docker-compose run --rm app jsonnet "${@//src\//}" --ext-str EDXAPP_DATASOURCE=${EDXAPP_DATASOURCE_DEMO:-edx_app}
```
Comment (jmaupetit) [inline bin/jsonnet:16]:
```suggestion
DOCKER_USER=${DOCKER_USER} docker-compose run --rm app jsonnet "${@//src\//}" --ext-str EDXAPP_DATASOURCE="${EDXAPP_DATASOURCE}"
```

File: Makefile [hunk 6]
```diff
@@ -78,16 +78,26 @@ down: ## remove stack (warning: it removes the database container)
...
+	@echo "Wait for mysql to be up..."
+	@$(WAIT_MYSQL)
+.PHONY: fixtures-mysql
+
+fixtures: \
+	run \
+	fixtures-es \
+	fixtures-mysql
+	eval 'scripts/users-permissions.sh grafana'
+fixtures: ## Load test data (for development)
```
Comment (jmaupetit) [inline Makefile:100]: Is this required for this feature? Also I'm not sure to understand why you need to `eval` here (and there is a block indentation issue).

**Review by jmaupetit — COMMENTED** [review]
Why not adding a `compile-demo` Makefile target that sets the `EDXAPP_DATASOURCE` environment variable and allows to switch from one data source to the other?

File: scripts/users-permissions.sh (hunk 2 above)
Comment (quitterie-lcs) [reply scripts/users-permissions.sh:48]: Not anymore indeed

File: Makefile (hunk 6 above)
Comment (quitterie-lcs) [reply Makefile:100]: Not necessary anymore

File: Makefile [hunk 7]
```diff
@@ -32,7 +32,7 @@ var/lib/grafana:
 
 var/lib/grafana/%.json: src/%.jsonnet
 	mkdir -p $(shell dirname $@)
-	bin/jsonnet -o /$@ $<
+	bin/jsonnet --ext-str EDXAPP_DATASOURCE=$${EDXAPP_DATASOURCE_DEMO:-edx_app} -o /$@ $<
```
Comment (jmaupetit) [inline Makefile:35]:
```suggestion
	bin/jsonnet --ext-str EDXAPP_DATASOURCE=$(EDXAPP_DATASOURCE) -o /$@ $<
```

File: Makefile [hunk 8]
```diff
@@ -70,6 +70,11 @@ compile: \
 compile: ## compile jsonnet sources to json
 .PHONY: compile
 
+compile-demo:
+    EDXAPP_DATASOURCE_DEMO="edx_app_demo" $(MAKE) -B compile
```
Comment (jmaupetit) [inline Makefile:74]:
```suggestion
    EDXAPP_DATASOURCE="edx_app_demo" $(MAKE) -B compile
```

**Comment by jmaupetit** [comment]
Adding this data source forces us to make requests both compatible with MySQL and SQLite. We've decided that it was a little bit too much for a public demo of the tool.

**Pull request closed without merging**